Formula/o/odin.rb

```diff
@@ -18,18 +18,112 @@ class Odin < Formula
+    end
+
+    resource "raylib" do
+      url "https://github.com/raysan5/raylib/archive/refs/tags/5.0.tar.gz"
+      sha256 "98f049b9ea2a9c40a14e4e543eeea1a7ec3090ebdcd329c4ca2cf98bc9793482"
+    end
+
+    resource "raygui" do
+      url "https://github.com/raysan5/raygui/archive/refs/tags/4.0.tar.gz"
+      sha256 "299c8fcabda68309a60dc858741b76c32d7d0fc533cdc2539a55988cee236812"
+    end
+  end
+
   fails_with gcc: "5" # LLVM is built with GCC
 
   def install
     llvm = deps.map(&:to_formula).find { |f| f.name.match?(/^llvm(@\d+(\.\d+)*)?$/) }
 
+    # Delete pre-compiled binaries which brew does not allow.
+    system "find", "vendor",
+                   "(",
+                     "-name", "*.lib",   "-o",
+                     "-name", "*.dll",   "-o",
+                     "-name", "*.a",     "-o",
+                     "-name", "*.dylib", "-o",
+                     "-name", "*.so.*",  "-o",
+                     "-name", "*.so",
+                   ")",
+                   "-delete"
+    if OS.mac?
+      # Compile glfw.
+      resource("glfw").stage do
+        system "cmake", "-S", ".", "-B", "build",
+          "-DGLFW_BUILD_EXAMPLES=OFF",
+          "-DGLFW_BUILD_TESTS=OFF",
+          "-DGLFW_BUILD_DOCS=OFF",
+          *std_cmake_args
+
+        cd "build" do
+          system "make"
+          cp "src/libglfw3.a", buildpath/"vendor/glfw/lib/darwin/libglfw3.a"
+        end
+      end
+
+      # Compile miniaudio.
+      cd buildpath/"vendor/miniaudio/src" do
+        system "make"
+      end
+
+      # Compile raylib.
+      raylib_buildpath = buildpath/"raylib"
+      raylib_buildpath.install resource("raylib")
+
+      raylib_installpath = if Hardware::CPU.arm?
+        "vendor/raylib/macos-arm64"
+      else
+        "vendor/raylib/macos"
+      end
+
+      cd raylib_buildpath/"src" do
+        # static.
+        system "make", "PLATFORM=PLATFORM_DESKTOP"
+        cp "libraylib.a", buildpath/raylib_installpath
+
+        # dynamic.
+        system "make", "PLATFORM=PLATFORM_DESKTOP", "RAYLIB_LIBTYPE=SHARED"
+        (buildpath/raylib_installpath).install Pathname.glob(raylib_buildpath/"src/*.dylib")
+      end
+
+      # Compile raygui.
+      resource("raygui").stage do
+        cp "src/raygui.h", "src/raygui.c"
+
+        # static.
+        system ENV.cc, "-c", "-o", "raygui.o", "src/raygui.c",
+          "-fpic", "-DRAYGUI_IMPLEMENTATION", "-I#{raylib_buildpath}/src"
+        system "ar", "-rcs", "libraygui.a", "raygui.o"
+        cp "libraygui.a", buildpath/raylib_installpath
+
+        # dynamic.
+        system ENV.cc, "-o", "libraygui.dylib", "src/raygui.c",
+          "-shared", "-fpic", "-DRAYGUI_IMPLEMENTATION", "-framework", "OpenGL",
+          "-lm", "-lpthread", "-ldl",
+          "-I#{raylib_buildpath}/src", "-L#{buildpath/raylib_installpath}", "-lraylib",
+          "-Wl,-rpath,#{libexec/raylib_installpath}"
+        cp "libraygui.dylib", buildpath/raylib_installpath
```

Comment:
Why do you need a static and dynamic version of the same library?

Reply:
See reply under the same comment above